iothub/device/tests/Mqtt/MqttIotHubAdapterTest.cs

```diff
@@ -37,6 +39,32 @@ public void TestPopulateMessagePropertiesFromPacket_NormalMessage()
             Assert.AreEqual("MessageId1", message.SystemProperties["message-id"]);
         }
 
+        [TestMethod]
+        [ExpectedException(typeof(MessageTooLargeException), AllowDerivedTypes = false)]
+        public void TestPopulateMessagePropertiesFromMessage_TopicNameTooLong()
+        {
+            var message = new Message();
+
+            for (int i = 0; i < 65; i++)
+            {
+                message.Properties.Add(Guid.NewGuid().ToString(), new string('1', 1024));
+            }
+
+            var publishPacket = new PublishPacket(QualityOfService.AtMostOnce, false, false)
```

Comment:
Is it better to wrap this common code in a method? I see this used in unit test above this as well.